### ISSUE TITLE
sendMessage -> payload.author.id error

### ISSUE THREAD

**Issue by Vicg853** [discordeno/discordeno#1889]
**To Reproduce** Steps to reproduce the behavior:

1. When using the sendMessage() function
2. Returns...
![image](https://user-images.githubusercontent.com/37887254/148538551-baa8552c-4920-4bdd-8ad7-fab988a4827a.png)

**Version details (please complete the following information):**

- Discordeno version: [e.g. 13.0.0-rc18]
- Deno version: [e.g. 1.17.1]

**Additional context** Add any other context about the problem here.
I did some digging... when calling sendMessage(), a message transformer is called (```https://deno.land/x/discordeno@13.0.0-rc18/src/transformers/message.ts```) and it seems that in its 18th line is called, looking for a payload.author.id
but when console logging the payload there is neither an author property nor even a payload property. 

Is it just maybe me using it with the wrong Discord permission maybe or it really isn't supposed to contain this property. 
Discord permissions on the server, currently include administration and either bot/gateway intents contain: ```"Guilds", "GuildMessages", 'DirectMessages'```

**Comment by Skillz4Killz** [comment]
Do you have any way to replicate this behavior? For example if you can send me a `mod.ts` file with a createBot and such that i can simply run and test this i can debug it. Sadly i am not being able to replicate this error.

**Comment by Vicg853** [comment]
Sure!

I pushed all here: https://github.com/Disq-Code-Bot/disq-code-replic 

If you prefer I may send it here directly too...

**Comment by lts20050703** [comment]
The error seems to be the discord payload doesn't have the property `author`, which is impossible?

**Comment by lts20050703** [comment]
or it could be discord payload is undefined/null (aka discord payload return NOTHING, instead of a proper message payload)

**Comment by Vicg853** [comment]
So you couldn't replicate the error? (just so I can know, because then probably there something I'm doing wrong somewhere) 

Ok. I'm going to fork the project then, to try and debug this with some console logs or something else.  

I'll send the results (or solution if I find one).

**Comment by itohatweb** [comment]
I am not able to replicate this error.

This error also seems to be very odd, since the `author` property of a `message` should always exist when using the `sendMessage` function.

**Comment by Vicg853** [comment]
Figured out what it is. It seems there are two issues here:

- Requests made to the Rest listener by the rest.runMethod method that receive 401 errors aren't handled; 
- The second issue: I discovered that the secret was the issue, which already was something weird as I'm testing it by only using "1234" as a secret (that by the way, yes, they are correctly defined both in the gateway and rest). But... I still tried changing the secret and also leaving it undefined. And after looking into it, it seems that, independently of what is set as secret, the rest.runMethod method, makes its call to the Rest API with the following "discordeno_best_lib_ever" as value for the secret. I haven't looked after the origin of this, but after hard resetting this value to "1234" just before the call, this was solved.

After this, the bot worked just fine.

I added a little handler for API errors so that at least the error is handled correctly, instead of getting undefined when the payload is equal to ```{ error: "Invalid authorization key." }```. #1893 

But haven't looked into the secret definition issue yet. But this is maybe a mistyping error and I think you know where this resides.
